cli/tools/registry/pm.rs

```diff
@@ -456,3 +513,53 @@ fn update_config_file_content(
   .map(|formatted_text| formatted_text.unwrap_or_else(|| new_text.clone()))
   .unwrap_or(new_text)
 }
+
+#[cfg(test)]
+mod test {
+  use deno_semver::VersionReq;
+
+  use super::*;
+
+  #[test]
+  fn test_parse_add_package_req() {
+    assert_eq!(
+      AddPackageReq::parse("jsr:foo").unwrap(),
+      AddPackageReq {
+        alias: "foo".to_string(),
+        value: AddPackageReqValue::Jsr(PackageReq::from_str("foo").unwrap())
+      }
+    );
+    assert_eq!(
+      AddPackageReq::parse("alias@jsr:foo").unwrap(),
+      AddPackageReq {
+        alias: "alias".to_string(),
+        value: AddPackageReqValue::Jsr(PackageReq::from_str("foo").unwrap())
+      }
+    );
+    assert_eq!(
+      AddPackageReq::parse("@alias/pkg@npm:foo").unwrap(),
+      AddPackageReq {
+        alias: "@alias/pkg".to_string(),
+        value: AddPackageReqValue::Npm(PackageReq::from_str("foo").unwrap())
+      }
+    );
+    assert_eq!(
+      AddPackageReq::parse("@alias/pkg@jsr:foo").unwrap(),
+      AddPackageReq {
+        alias: "@alias/pkg".to_string(),
+        value: AddPackageReqValue::Jsr(PackageReq::from_str("foo").unwrap())
+      }
+    );
```

Comment:
What about `alias@jsr:foo@^1.5.0`? Is it handled here? If not can you add a test that expects it to fail?

Reply:
That will succeed and then fail on download. I added a test for it succeeding.

Reply:
Why will it fail on download?

Reply:
Because it's not a scoped package.